### PR TITLE
enum: All enum serialized to lower case

### LINKED ISSUE #58: All enum should be serialized to snake case instead of camel case

### PR DESCRIPTION
**Pull request by victorjdg**
Used serde to rename all variants to lowercase.

Signed-off-by: Víctor Javier Díaz Garrido <vicdigar@hotmail.com>

### CODE REVIEW

File: src/cli/npc.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ impl Display for CliError {
     }
 }
 
+#[serde(rename_all = "lowercase")]
 enum CliResult {
     Full(NetState),
     Ifaces(Vec<Iface>),
```
Comment (ffmancera) [inline src/cli/npc.rs]: I think this rename is not needed as this enum is not being serialize. What do you think? @cathay4t

File: src/cli/npc.rs [hunk 2]
```diff
@@ -27,6 +28,7 @@ enum CliResult {
     NisporError(NisporError),
 }
 
+#[serde(rename_all = "lowercase")]
 enum CliOutputType {
     Json,
     Yaml,
```
Comment (ffmancera) [inline src/cli/npc.rs]: Same here.

File: src/varlink/info_nispor.rs [hunk 3]
```diff
@@ -10,6 +10,7 @@ use varlink::{self, CallTrait};
 
 #[allow(dead_code)]
 #[derive(Clone, PartialEq, Debug)]
+#[serde(rename_all = "lowercase")]
 pub enum ErrorKind {
     Varlink_Error,
```
Comment (ffmancera) [inline src/varlink/info_nispor.rs]: Same here.

**Review by ffmancera — CHANGES_REQUESTED** [review]
Hello! Thanks for your contributions. I've made some comments below. Please take a look, thanks!

File: src/cli/npc.rs (hunk 1 above)
Comment (victorjdg) [reply src/cli/npc.rs]: Ok, i'll make the changes.

File: src/cli/npc.rs (hunk 2 above)
Comment (victorjdg) [reply src/cli/npc.rs]: Done.

File: src/varlink/info_nispor.rs (hunk 3 above)
Comment (victorjdg) [reply src/varlink/info_nispor.rs]: Done.

**Review by ffmancera — APPROVED** [review]

File: src/lib/error.rs [hunk 4]
```diff
@@ -3,6 +3,7 @@ use rtnetlink;
 use serde_derive::Serialize;
 
 #[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "lowercase")]
 pub enum ErrorKind {
```
Comment (cathay4t) [inline src/lib/error.rs:7]: ErrorKind does not needed to be lower case as this is rust only.

File: src/lib/ifaces/bond.rs [hunk 5]
```diff
@@ -356,6 +358,7 @@ impl From<u8> for BondSubordinateState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```
Comment (cathay4t) [inline src/lib/ifaces/bond.rs]: Please also use `#[serde(rename = "link_up")]`

File: src/lib/ifaces/bridge.rs [hunk 6]
```diff
@@ -8,6 +8,7 @@ use serde_derive::{Deserialize, Serialize};
 use std::collections::HashMap;
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
 pub enum BridgeStpState {
     Disabled,
     KernelStp,
```
Comment (cathay4t) [inline src/lib/ifaces/bridge.rs:14]: Please also use `#[serde(rename = "kernel_stp")]`

File: src/lib/ifaces/bridge.rs [hunk 7]
```diff
@@ -32,6 +33,7 @@ impl From<u32> for BridgeStpState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
 pub enum BridgeVlanProtocol {
     #[serde(rename = "802.1Q")]
```
Comment (cathay4t) [inline src/lib/ifaces/bridge.rs:38]: Changed to `#[serde(rename = "802.1q")]`

File: src/lib/ifaces/bridge.rs [hunk 8]
```diff
@@ -184,6 +187,7 @@ impl From<u8> for BridgePortStpState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
 pub enum BridgePortMulticastRouterType {
     Disabled,
     TempQuery,
```
Comment (cathay4t) [inline src/lib/ifaces/bridge.rs:193]: Please use `#[serde(rename = "temp_query")]`

File: src/lib/ifaces/iface.rs [hunk 9]
```diff
@@ -65,6 +66,7 @@ impl Default for IfaceType {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```
Comment (cathay4t) [inline src/lib/ifaces/iface.rs]: Please use `#[serde(rename = "lower_layer_down")]`

File: src/lib/ifaces/iface.rs [hunk 10]
```diff
@@ -81,6 +83,7 @@ impl Default for IfaceState {
...
 pub enum IfaceFlags {
```
Comment (cathay4t) [inline src/lib/ifaces/iface.rs:87]: Please use `#[serde(rename = "all_multi")]`, or maybe a better name.

**Review by cathay4t — CHANGES_REQUESTED** [review]
You cannot just simply put `#[serde(rename_all = "lowercase")]` on every enum or struct.

Please note:
 * All CameCase should convert to `abc_def` format.

**Comment by cathay4t** [comment]
Thanks for the contribution!

Please be aware I intend to include this PR in the 1.0 release which is scheduled at the end of this week.

**Comment by ffmancera** [comment]
> You cannot just simply put `#[serde(rename_all = "lowercase")]` on every enum or struct.
> 
> Please note:
> 
>     * All CameCase should convert to `abc_def` format.

Oh, the issue was saying to lower case not snake case. In that case, renaming all the enum members will be needed. Thanks for pointing it out.

**Comment by victorjdg** [comment]
@cathay4t I used `#[serde(rename_all = "snake_case")]` to get the `abc_def` format.

I tested it using the `npc` command and in the interface flags, the enum contains `LowerUp` and its showing `lower_up`.

File: src/lib/ifaces/iface.rs [hunk 11]
```diff
@@ -65,6 +66,8 @@ impl Default for IfaceType {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename = "lower_layer_down")]
```
Comment (ffmancera) [inline src/lib/ifaces/iface.rs]: That is not needed as you are using snake_case on rename_all.

File: src/lib/ifaces/iface.rs [hunk 12]
```diff
@@ -81,6 +84,8 @@ impl Default for IfaceState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename = "all_multi")]
```
Comment (ffmancera) [inline src/lib/ifaces/iface.rs]: That is not needed as you are using snake_case on rename_all.

File: src/lib/ifaces/bond.rs [hunk 13]
```diff
@@ -356,6 +358,8 @@ impl From<u8> for BondSubordinateState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename = "link_up")]
```
Comment (ffmancera) [inline src/lib/ifaces/bond.rs]: That is not needed as you are using snake_case on rename_all.

File: src/lib/ifaces/bridge.rs [hunk 14]
```diff
@@ -8,6 +8,8 @@ use serde_derive::{Deserialize, Serialize};
 use std::collections::HashMap;
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename = "kernel_stp")]
```
Comment (ffmancera) [inline src/lib/ifaces/bridge.rs]: That is not needed as you are using snake_case on rename_all.

File: src/lib/ifaces/bridge.rs [hunk 15]
```diff
@@ -32,6 +34,8 @@ impl From<u32> for BridgeStpState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename = "802.1q")]
```
Comment (ffmancera) [inline src/lib/ifaces/bridge.rs]: That is not needed as you are using snake_case on rename_all.

File: src/lib/ifaces/bridge.rs [hunk 16]
```diff
@@ -184,6 +189,8 @@ impl From<u8> for BridgePortStpState {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename = "temp_query")]
```
Comment (ffmancera) [inline src/lib/ifaces/bridge.rs]: That is not needed as you are using snake_case on rename_all.

**Review by ffmancera — CHANGES_REQUESTED** [review]

**Comment by ffmancera** [comment]
I've noticed that the `BondMode` enum has some renames already but using `-` instead of `_` . Should we rename them too? @cathay4t

**Comment by cathay4t** [comment]
> I've noticed that the `BondMode` enum has some renames already but using `-` instead of `_` . Should we rename them too? @cathay4t

No. Those are using the term of kernel sysfs.

**Review by ffmancera — APPROVED** [review]

**Review by cathay4t — APPROVED** [review]
This is better than I expected. Thank you.

**Pull request merged**